pkg/diag/validator/pod.go

```diff
@@ -386,12 +388,15 @@ func trimSpace(msg string) string {
 	return strings.Trim(msg, " ")
 }
 
-func getPodLogs(po *v1.Pod, c string) []string {
+func getPodLogs(po *v1.Pod, c string, sc proto.StatusCode) (proto.StatusCode, []string) {
 	logrus.Debugf("Fetching logs for container %s/%s", po.Name, c)
 	logCommand := []string{"kubectl", "logs", po.Name, "-n", po.Namespace, "-c", c}
 	logs, err := runCli(logCommand[0], logCommand[1:])
 	if err != nil {
-		return []string{fmt.Sprintf("Error retrieving logs for pod %s. Try `%s`", po.Name, strings.Join(logCommand, " "))}
+		return sc, []string{fmt.Sprintf("Error retrieving logs for pod %s. Try `%s`", po.Name, strings.Join(logCommand, " "))}
```

Comment:
Can we include `err` in this message?  Or log it?  We're throwing away information otherwise.

Reply:
done and tests updated